src/player.cpp

```diff
@@ -7599,6 +7620,148 @@ void player::hardcoded_effects(effect &it)
                 }
             }
         }
+    } else if ( id == "skin_rot" || id == "skin_rot_recovery") {
+        bool recovery = id == "skin_rot_recovery" ? true : false;
+        float heal_chance = 7200.0;
+        float spread_chance = 3600.0;
+        float health = get_healthy();
+        float health_mod = health / 100.0 + 1.0;
+
+        //First, try to spread the disease into a conected body part
+        if (health > 0) {
+            spread_chance *= health_mod;
+        } else if (health < 0) {
+            spread_chance /= health_mod;
+        }
+        if (!recovery && one_in( (int)ceil( spread_chance ) ) ) {
+            switch (bp) {
```

Comment:
That switch is huge. You could easily cut down the number of lines it by adding a helper lambda method like this:
```
const auto try_infect = [this]( body_part bp ) {
    if( !has_effect( "skin_rot", bp ) ) {
        add_effect( "skin_rot", 1, bp );
    }
};
```

And then just invoke it in the switch.
You could even make it a tiny bit more complex and have it do the rng part (to avoid nesting switches).

Also, it would be a good idea not to reinfect recovering parts, otherwise you may get an infinite infection loop with many parts being infected and recovering at the same time.